torch/csrc/cuda/nccl.cpp

```diff
@@ -659,7 +730,11 @@ void all2all_single_equal_split(
 #if defined(USE_ROCM) && ROCM_VERSION >= 50000
   NCCL_CHECK(ncclAllToAll(sendbuff, recvbuff, count, type, comm, stream));
 #else
-  NCCL_CHECK(ncclCommCount(comm, &numranks));
+  if (!comm_nonblocking) {
+    NCCL_CHECK(ncclCommCount(comm, &numranks));
+  } else {
+    NCCL_CHECK_NONBLOCKING(ncclCommCount(comm, &numranks), _comm);
```

Comment:
Might be unnecessary to also do a non-blocking check for `ncclCommCount` (unsure if there exists documentations on exactly which API calls might leave a communicator in an in-progress state).